### PR TITLE
Added Google authentication

### PR DESCRIPTION
**Pull request by subodhk01**
As soon as user logins, if theres no Student object assigned to a new Student object gets created for the User with only name as the filled entry, rest all entries would be added on the profile page. The only problem currently is how to assign department to the Student Object as it cannot be changed on the profile page.

### CODE REVIEW

File: myapp/views.py [hunk 1]
```diff
@@ -98,14 +99,24 @@ def index(request):
...
-    if request.method == 'GET':
-        context = {"user": UsrObj}
-        return render(request, 'myapp/profile.html', context)
+    email = request.user.email
+    if not(re.findall("15@", email) or re.findall("16@", email)):
+        return redirect('logout')
+    try:
+        # add image field edit
+        u = request.user
+        UsrObj = Student(name=u.student.name, department=u.student.department,
```
Comment (shivanshs9) [inline myapp/views.py:108]: Can't see why you're creating a local instance of Student here. Shouldn't you be retrieving the Student record linked with the logged in user?

Comment (shivanshs9) [reply myapp/views.py:108]: And if the user will change profile data, then POST method should be handled to update the retrieved Student record data.

Comment (subodhk01) [reply myapp/views.py:108]: That code was already there, I added the expect: part, for when the user logins for the first time, to create a new Student object.

File: myapp/views.py [hunk 2]
```diff
@@ -98,50 +99,55 @@ def index(request):
...
-        if not (u.student.genPic1.name.lower().endswith(('.png', '.jpg', '.jpeg'))):
-            return render(request, 'myapp/profile.html', {"user": UsrObj, "image": "Image should be in .png, .jpg or .jpeg format"})
-        extension = u.student.genPic1.name.lower()[u.student.genPic1.name.lower().rfind("."):] 
+        u.genPic1 = request.FILES.get('genPic1')
+        if not (u.genPic1.name.lower().endswith(('.png', '.jpg', '.jpeg'))):
+            return render(request, 'myapp/profile.html', {"user": u, "image": "Image should be in .png, .jpg or .jpeg format"})
+        extension = u.genPic1.name.lower()[u.genPic1.name.lower().rfind("."):] 
 
-        u.student.genPic1.name = u.username + extension
+        request.user.student.genPic1.name = request.user.username + extension
```
Comment (shivanshs9) [inline myapp/views.py:132]:
```suggestion
        u.genPic1.name = request.user.username + extension
```

File: myapp/views.py [hunk 3]
```diff
@@ -129,7 +132,7 @@ def profile(request):
             return render(request, 'myapp/profile.html', {"user": u, "image": "Image should be in .png, .jpg or .jpeg format"})
         extension = u.genPic1.name.lower()[u.genPic1.name.lower().rfind("."):] 
 
-        request.user.student.genPic1.name = request.user.username + extension
+        u.genPic1.name = request.user.username + extension
```
Comment (shivanshs9) [inline myapp/views.py:135]: Now this line will crash with **NameError** when POST request is made and Student doesn't exists.

Comment (subodhk01) [reply myapp/views.py:135]: If a legal POST request is made, it means that there is for sure a user.

**Pull request merged**